src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/CreateClaimCallbackHandler.java

```diff
@@ -118,11 +118,15 @@ private CallbackResponse issueClaim(CallbackParams callbackParams) {
         CaseData caseData = caseDetailsConverter.toCaseData(callbackParams.getRequest().getCaseDetails());
         String referenceNumber = referenceNumberRepository.getReferenceNumber();
 
+        Party respondent = updatePartyWithPartyName(caseDetails.getData().get(RESPONDENT));
+        Party applicant = updatePartyWithPartyName(caseDetails.getData().get(CLAIMANT));
         CaseDocument sealedClaim = sealedClaimFormGenerator.generate(
             caseData.toBuilder()
                 .claimIssuedDate(issueDate)
                 .legacyCaseReference(referenceNumber)
                 .claimSubmittedDateTime(submittedAt)
+                .applicant1(applicant)
+                .respondent1(respondent)
                 .build(),
```

Comment:
hmm I won't stop as it fixes, but I don't like that we pass in this caseData.toBuilder() stuff. In my mind the generate method should map case details -> case data and then it has the values available. The generate method could then take in case details and a date (in this case submittedAt - which again I think makes sense as all documents will need a generated date). Then this code becomes a lot simpler

Reply:
I think the transform of respondent and claimant could happen in the case data convertor too, then it's avaiable in the generate method and can be added to data with data.put(RESPONDENT, case.getRespondent);

Reply:
made a tech debt ticket here https://tools.hmcts.net/jira/browse/CMC-750